app/auth/config.go

```diff
@@ -0,0 +1,29 @@
+package auth
+
+import (
+	"x-gwi/app/x/env"
+)
+
+type ConfigAuth struct {
+	AppRoot *BasicAuth
+}
+
+type BasicAuth struct {
+	UserName string
+	PassWord string
+	PassHash string
+}
+
+func NewConfigAuth() *ConfigAuth {
+	return &ConfigAuth{
+		AppRoot: &BasicAuth{
+			UserName: env.Env("AUTH_APPROOT_USERNAME", "app-root"),
+			PassWord: env.Env("AUTH_APPROOT_PASSWORD", ""),
+			PassHash: env.Env("AUTH_APPROOT_PASSHASH", "$2a$08$rvzL6XK2K/UPnK1fkHS3TeDWyc9nbYMAnZBSkiiWCamNilcYxA4jK"),
```

Comment:
What is this for?

Reply:
app/auth - is a starting point for authentication and authorization functionality; checking is not implemented but if required it can be turned on on app/env level;

Reply:
Yes, but I meant the pass hash specifically, why is it in config? What is it used for?

Reply:
_PASSHASH in config is to simplify potential emulation of authenticator to be able to start without external storage.
It is not recommended to save passwords in config so to be able to check if password is env _PASSWORD is the same as hardcoded one there is common methodology to store not password value but secure king of hash of the value like bcrypt.
_PASSHASH is here to distinguish in potential authenticator provided _PASSWORD vs _PASSHASH to be checked against to
If part of functionality depends on authenticated admin~app-root we don't have to relay on external storage but we can validate env _PASSWORD against env _PASSHASH with default value set in code (or set in code as const plus one from env).

Does it answer your question?

Reply:
It does, thank you.